From 2fdff5aac99b34dd68920297e80d9b511d9ebd6e Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Thu, 31 Oct 2019 15:58:24 +0530
Subject: [PATCH 0322/1921] octeontx2-pf: Allow setting VF mac and vlan only
 when PF is UP

To install NPC MCAM rules to support VF MAC and VF VLAN settings
PF needs to know ingress hw channel number. This info is sent to
PF by AF in response to NIXLF_ALLOC mbox ie NIXLF init request.

So skip 'set vf vlan' and 'set vf mac' if PF interface is not UP.

Change-Id: Ib8d0c5c84296aa22090d39038055e769668ac07f
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18079
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index cbb4b4ec7059..a425ebcbd4b8 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1987,6 +1987,9 @@ static int otx2_set_vf_mac(struct net_device *netdev, int vf, u8 *mac)
 	struct pci_dev *pdev = pf->pdev;
 	struct otx2_vf_config *config;
 
+	if (!netif_running(netdev))
+		return -EAGAIN;
+
 	if (vf >= pci_num_vf(pdev))
 		return -EINVAL;
 
@@ -2034,6 +2037,9 @@ static int otx2_set_vf_vlan(struct net_device *netdev, int vf, u16 vlan, u8 qos,
 	struct pci_dev *pdev = pf->pdev;
 	struct otx2_vf_config *config;
 
+	if (!netif_running(netdev))
+		return -EAGAIN;
+
 	if (vf >= pci_num_vf(pdev))
 		return -EINVAL;
 
-- 
2.31.1

